virtual-display-driver-cli/src/main.rs

```diff
@@ -0,0 +1,343 @@
+use clap::Parser;
+use client::Client;
+use joinery::JoinableIterator;
+use lazy_format::lazy_format;
+use owo_colors::OwoColorize;
+use serde::{Deserialize, Serialize};
+
+mod client;
+
+#[derive(Debug, Parser)]
+struct Args {
+    /// Format output as JSON.
+    #[clap(short, long)]
+    json: bool,
+
+    #[clap(subcommand)]
+    command: Command,
+}
+
+#[derive(Debug, Parser)]
+enum Command {
+    /// List currently connected virtual monitors.
+    List,
+    /// Add a new virtual monitor.
+    Add {
+        /// Width of the virtual monitor.
+        width: driver_ipc::Dimen,
+
+        /// Height of the virtual monitor.
+        height: driver_ipc::Dimen,
+
+        /// More resolutions to add as extra modes for the virtual monitor.
+        more_widths_and_heights: Vec<driver_ipc::Dimen>,
+
+        /// Refresh rate of the virtual monitor. Pass multiple times to
+        /// support multiple refresh rates.
+        #[clap(short, long, default_value = "60")]
+        refresh_rate: Vec<driver_ipc::RefreshRate>,
+
+        /// Manual ID to set for the monitor. Must not conflict with an
+        /// existing virtual monitor's ID.
+        #[clap(long)]
+        id: Option<driver_ipc::Id>,
+
+        /// Optional label for the virtual monitor.
+        #[clap(long)]
+        name: Option<String>,
+
+        /// Set the virtual monitor to disabled on creation.
+        #[clap(long)]
+        disabled: bool,
+    },
+    /// Add a new resolution/refresh rate mode to an existing virtual monitor.
+    AddMode {
+        /// ID of the virtual monitor to add a mode to.
+        id: driver_ipc::Id,
+
+        /// Width of the new mode.
+        width: driver_ipc::Dimen,
+
+        /// Height of the new mode.
+        height: driver_ipc::Dimen,
+
+        /// Refresh rate for the new mode. Pass multiple times to support
+        /// multiple refresh rates in this mode.
+        #[clap(short, long, default_value = "60")]
+        refresh_rate: Vec<driver_ipc::RefreshRate>,
+    },
+    /// Remove a resolution/refresh rate mode to an existing virtual monitor.
+    RemoveMode {
+        /// ID of the virtual monitor to add a mode to.
+        id: driver_ipc::Id,
+
+        /// The index of the mode to remove.
+        mode_index: usize,
+    },
+    /// Enable a virtual monitor.
+    Enable { id: driver_ipc::Id },
+    /// Disable one or more virtual monitors.
+    Disable { id: driver_ipc::Id },
+    /// Remove one or more virtual monitors.
+    Remove { id: Vec<driver_ipc::Id> },
+    /// Remove all virtual monitors.
+    RemoveAll,
+}
+
+fn main() -> eyre::Result<()> {
+    let args = Args::parse();
+
+    match args.command {
+        Command::List => {
+            let mut client = Client::connect()?;
```

Comment:
Regarding the comment about list calls and moving things up. We can probably just move this connection call up into main so all enum variants have access to one instance (we don't have to call this in every match arm as a result).

Reply:
Yeah, that would definitely make more sense!

Reply:
_Marking as resolved, this will be handled in a follow-up_